### DISCUSSION TITLE
rust-course/practice/third-party-libs

### DISCUSSION THREAD

**Discussion by sunface** [sunface/rust-course#615] [章节评论区]
https://course.rs/practice/third-party-libs.html

**Comment by ZhuJiaqi9905** [comment]
crossbeam也值得拥有姓名

**Reply by zhwei820** [reply]
// MPSC channels are built as a wrapper around MPMC channels, which
// were ported from the `crossbeam-channel` crate. MPMC channels are
// not exposed publicly, but if you are curious about the implementation,
// that's where everything is.

https://github.com/rust-lang/rust/pull/93563/files#diff-9d743baec8192970666e3beb38f5e07b7caf6891f3ec5d15f126bbee2f3ec28a

**Comment by sunface** [comment]
> crossbeam也值得拥有姓名

收到～～

**Comment by chongyi** [comment]
ORM 可以添加 [sea-orm](https://github.com/SeaQL/sea-orm)，这个库未来可期，整体设计也比较优秀

**Comment by Roxannej** [comment]
前端的一些类似SWC的打包工具啥的也可以加进去吗

**Comment by paulllb** [comment]
有没有本地缓存的库推荐

**Comment by zhng1456** [comment]
因为MeiliSearch接触到的Rust，还在持续学习中

**Comment by itfanr** [comment]
slog 确实不好用，不推荐

**Comment by Jia851747508** [comment]
有支持oracle的orm框架吗？

**Comment by CherishCai** [comment]
服务发现，Nacos 社区同学支持的 https://github.com/nacos-group/nacos-sdk-rust
从「圣经」学来的 Rust 实践

**Comment by SoonIter** [comment]
indicatif 这样的cli常用库可以归一个类感觉

**Comment by bingo789** [comment]
找不到数据结构与算法的模块，rust还是年轻

**Reply by Jiewyjson** [reply]
数据结构感觉可以直接看std,算法的话直接刷刷力扣或者直接上codeforces也可以吧？cf上面有用rust打比赛的大神，模板帖子也有

**Comment by zhouyu00** [comment]
有没有推荐新手阅读的rust源码

**Comment by Jia851747508** [comment]
Excel读写有什么好用的库吗？

**Comment by photino** [comment]
Web框架能不能推荐一下我们开发的[Zino](https://zhuanlan.zhihu.com/p/621580693)：开箱即用的企业级应用开发框架

**Reply by ly2020888** [reply]
支持一下

**Comment by zrll12** [comment]
能不能教一下ffi绑定怎么在windows下编译

**Comment by BlackAdvancedCat** [comment]
有科学计算需求的话，有以下库可以考虑，ndarray（多维数组），nalgebra（线性代数），sprs/nalgebra_sparse（稀疏矩阵）

**Reply by BlackAdvancedCat** [reply]
ndarray自带rayon

**Comment by RenMinZhang** [comment]
sql server 没有第3方库吗

**Comment by nmxkli18** [comment]
salvo 也可以

**Comment by cj495840252** [comment]
Polars, 对标python的pandas，语法基本一样，也支持python调用

**Comment by 4innocent** [comment]
actix-web都已经 4.4了，axum还是 0.7.3，到底是 actix-web 问题太多发版太快，还是 axum 维护太懒😄

**Reply by liigo** [reply]
不要光看版本号

**Comment by onsalo** [comment]
有没有串口方面的

**Reply by ershuaili** [reply]
serialport-rs

**Comment by PeishuaiLee** [comment]
GUI方面的类库，也可以帮忙说明下哦。

**Reply by liigo** [reply]
iced, dioxus

**Comment by hiqishen** [comment]
还是用的时候去搜索大家用的哪个吧，不定啥时间就不更了或者有更好的了，rust官方。。。很多基础功能有但就是不全
